app/models/spree/order_decorator.rb

```diff
@@ -350,6 +350,17 @@ def pending_payments
     payments.select {|p| p.state == "checkout"} # Original definition
   end
 
+  def before_finalize_hook
+    line_item_variants.each do |variant| 
+      scoper.scope(variant)
+    end
+  end
+
+  def scoper
+    return @scoper unless @scoper.nil?
+    @scoper = OpenFoodNetwork::ScopeVariantToHub.new(distributor)
+  end
+
```

Comment:
this before_finalize_hook is what we would create in spree itself.
@sauloperez I was thinking we could merge this one with the one you have created "before_save_hook", that one (address_from_distributor) will work if placed at the top of the order.finalize! method which is where this scoping needs to be done.